### PR TITLE
[kueue] Wait for webhooks setup before ready.

### PR DESCRIPTION
**Pull request by trasc**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by trasc** [comment]
try 2.

/test pull-kueue-test-e2e-main-1-27
/test pull-kueue-test-e2e-main-1-28
/test pull-kueue-test-e2e-main-1-29

File: test/util/e2e.go [hunk 1]
```diff
@@ -65,11 +65,19 @@ func CreateVisibilityClient(user string) visibilityv1alpha1.VisibilityV1alpha1In
 	return visibilityClient
 }
 
-func KueueReadyForTesting(ctx context.Context, client client.Client) {
+func KueueReadyForTesting(ctx context.Context, c client.Client) {
```
Comment (mimowo) [inline test/util/e2e.go:68]: Can we delete this function if we are waiting for the "webooks setup before ready"?

Comment (trasc) [reply test/util/e2e.go:68]: no, this is running in the test binary

Comment (mimowo) [reply test/util/e2e.go:68]: but should not the test binary wait until Kueue is healthy / ready, rather than trying to periodically create the items? If we still periodically need to create items I don't seem much gain of delaying the healthy status.

Comment (trasc) [reply test/util/e2e.go:68]: It's just another way of checking the same thing.

I theory you can check if all the kueue-controller-manager pods are ready instead.

File: test/util/e2e.go [hunk 2]
```diff
@@ -65,11 +65,19 @@ func CreateVisibilityClient(user string) visibilityv1alpha1.VisibilityV1alpha1In
...
 	// To verify that webhooks are ready, let's create a simple resourceflavor
 	resourceKueue := utiltesting.MakeResourceFlavor("default").Obj()
-	gomega.Eventually(func() error {
-		return client.Create(context.Background(), resourceKueue)
+	gomega.EventuallyWithOffset(1, func() error {
+		return c.Create(ctx, resourceKueue)
 	}, StartUpTimeout, Interval).Should(gomega.Succeed())
-	ExpectResourceFlavorToBeDeleted(ctx, client, resourceKueue, true)
```
Comment (alculquicondor) [inline test/util/e2e.go:74]: why change this?

**Comment by alculquicondor** [comment]
A failure happened https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/1674/pull-kueue-test-e2e-main-1-28/1752369431784198144 :(

The webhook never became available.

File: test/util/e2e.go (hunk 2 above)
Comment (trasc) [reply test/util/e2e.go:74]: the reporting is hard to follow with that,  in case of multikueue you don't know which of the clusters is not ready.

File: test/util/e2e.go (hunk 1 above)
Comment (mimowo) [reply test/util/e2e.go:68]: I see, but this involves creating / deleting objects, something which users may not want to do in prod. I think it is better if e2e tests correspond better to production use. Also, the creation / deletion takes more time some may cover some user issues. If we go with this approach I would suggest to wait as for ready deployment, this can be done as here: 
```
kubectl wait --for=condition=available --timeout=3m deployment/kueue-controller-manager -n kueue-system
```
